front/lib/api/oauth/providers/base_oauth_stragegy_provider.ts

```diff
@@ -34,23 +61,30 @@ export interface BaseOAuthStrategyProvider {
     useCase: OAuthUseCase
   ) => boolean;
 
-  // If the provider has a method for getting the related credential, it must return a cleaned config.
-  getRelatedCredential?: (
+  // If the provider has a method for getting the related credential and/or if we need to update the config.
+  updateConfigAndGetRelatedCredential?: (
```

Comment:
The name is a bit misleading as it can only update the config right?

Shouldn't separate the two functions one to get the related credentials and another one to udpate the conifg (possibly based on the related credentials retrieved)?